plugins/ltac/rewrite.ml

```diff
@@ -1795,15 +1794,16 @@ let declare_an_instance n s args =
 
 let declare_instance a aeq n s = declare_an_instance n s [a;aeq]
 
-let anew_instance ~pstate atts binders (name,t) fields =
-  let program_mode = atts.program in
-  new_instance ~pstate ~program_mode atts.polymorphic
-    name binders t (Some (true, CAst.make @@ CRecord (fields)))
-    ~global:atts.global ~generalize:false Hints.empty_hint_info
+let anew_instance atts binders (name,t) fields =
+  let _id = Classes.new_instance atts.polymorphic
```

Comment:
@ejgallego for instance here we don't have to `assert (Option.is_empty pstate)` anymore (as there isn't a pstate option generated anymore), I do think it goes in the same direction as the original PR.